lib/stripe/oauth.rb

```diff
@@ -0,0 +1,92 @@
+module Stripe
+  module OAuth
+    class StripeOAuthClient < StripeClient
+      def self.default_client
+        @default_client ||= StripeOAuthClient.new(default_conn)
+      end
+
+      def execute_request(method, url,
+          api_base: nil, api_key: nil, headers: {}, params: {})
+        api_base ||= Stripe.connect_base
+        super(method, url, api_base: api_base, api_key: api_key, headers: headers, params: params)
+      end
+
+      def handle_api_error(http_resp)
+        begin
+          resp = StripeResponse.from_faraday_hash(http_resp)
+          type = resp.data[:error]
+          description = resp.data[:error_description]
+
+          unless type && type.is_a?(String)
+            raise StripeError.new("Indeterminate error")
+          end
+
+        rescue JSON::ParserError, StripeError
+          raise general_api_error(http_resp[:status], http_resp[:body])
+        end
+
+        error = OAuthError.new(
+          type, description,
+          http_status: resp.http_status, http_body: resp.http_body,
+          json_body: resp.data, http_headers: resp.http_headers
+        )
+
+        error.response = resp
+        raise(error)
+      end
+    end
+
+    module OAuthOperations
+      extend APIOperations::Request::ClassMethods
+
+      def self.request(method, url, params={}, opts={})
```

Comment:
Minor, but given that this is always called with a provided `params` and `opts`, and should never really be invoked by a user, what do you think about just making all params required?